docs/docs-graphql/schema.graphqls

```diff
@@ -3742,6 +3818,22 @@ The `ScalarTypeDefID` scalar type represents an identifier for an object of type
 """
 scalar ScalarTypeDefID
 
+type SearchResult {
+  filePath: String!
+
+  """A unique identifier for this SearchResult."""
+  id: SearchResultID!
+
+  lineNumber: Int!
```

Comment:
Any chance we could instead get:
- `start` / `end` (the raw position in the file, useful for byte slicing, etc)
- `startLine` / `endLine`
- `startColumn` / `endColumn`

Naming hard obviously, but feels like we should definitely have a way to get the start+end, and being able to get the columns feels useful too.

Maybe even a new `FilePosition` type might be useful here? For example, we already having something kinda similar in the `SourceMap` type (if we like the idea, I can update the `SourceMap` to have that shared type). That type could then be used for some cool things - e.g. you could have `File.read(start, end FilePosition)`.

Reply:
Having looked into it again, I'm still inclined to just wait until someone needs these more detailed APIs. The complexity arises in the fact that a line can actually have _many_ submatches, which means we would need a whole new type like `[]SearchMatches` or something to carry pairs of start/end offsets within the `MatchedLines`. `EndLine` can be deduced from `LineNumber` + counting linebreaks in `MatchedLines`. Searching for multiline strings is a bit of a fringe case so not sure I'd want to make the API favor it too much.